google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/AppEngineCloudType.java

```diff
@@ -202,6 +203,16 @@ public void setName(String name) {
         this.name = name;
       }
 
+      @Nullable
+      @Override
+      public File getFile() {
+        return new File(userSpecifiedFilePath);
```

Comment:
Regarding #560, even if the userSpecifiedFilePath is set, if that path does not end with either .jar or .war, then Dockerfile auto-generation will still produce an error. For this reason, I am not working on showing/hiding the custom deployment UI, but rather working on displaying an error message dialog.